patterns/MVC/README.md

```diff
@@ -0,0 +1,299 @@
+# MVC ( model-view-controller) Architecture
+
+In programming, model-view-controller (MVC) is an architectural design pattern that organizes an application's logic into distinct layers, each of which carries out a specific set of tasks. The layers also interact with each other to ensure that the application's functionality is delivered in a coordinated and consistent manner. The MVC methodology incorporates the entire application, from the user interface (UI) to the underlying data model. The three layers of MVC architecure are -
+
+- Model : Manages the application's data and business logic, including data validation, manipulation, and interaction with the database.
+- View : Presents the user interface to the user, rendering data received from the Model layer in a format suitable for presentation.
+- Controller : Interprets user input from the View layer, processes it, and interacts with the Model layer to retrieve or manipulate data accordingly.
+
+Sheldon, Robert. “What Is Model-View-Controller (MVC)?: Definition from TechTarget.” WhatIs, TechTarget, 12 Sept. 2023, [techtarget.com](www.techtarget.com/whatis/definition/model-view-controller-MVC#:~:text=In%20programming%2C%20model%2Dview%2D,a%20specific%20set%20of%20tasks)
+
+# Visual representation of MVC architecture
+
+![image](https://github.com/YogeshManni/pattern-library/assets/29475936/96056260-a719-4651-bee0-685ce6390af8)
+
+sadikarahmantanisha, "the-mvc-architecture", medium.com,19 Sept. 2023, [MVC architecture](https://medium.com/@sadikarahmantanisha/the-mvc-architecture-97d47e071eb2)
+
+---
+
+## Advantages of MVC
+
+- **Separation of Concerns**: MVC separates the application into distinct components, making it easier to understand, maintain, and modify.
+- **Modular Design**: Each component of MVC can be developed and tested independently, allowing for greater flexibility and reusability.
+- **Scalability**: MVC facilitates the scalability of the application by allowing developers to add new features or modify existing ones without affecting other parts of the application.
+
+---
+
+# MVC Architecture in Todo Task List Application
+
+For this issue, I will be explaining MVC architecture by writing code for to-do task list application. The Tech stack that I will be using is
+
+```
+Mongodb and Javascript - For Model part (Database)
+React - For View Part
+Node.js - For Controller part
+```
```

Comment:
A bulleted list and hyperlinks here is more appropriate than formatting this as code.